geopandas/geodataframe.py

```diff
@@ -289,10 +289,15 @@ def rename_geometry(self, col, inplace=False):
         geodataframe : GeoDataFrame
         """
         geometry_col = self.geometry.name
-        if not inplace:
-            return self.rename(columns={geometry_col: col}).set_geometry(col, inplace)
-        self.rename(columns={geometry_col: col}, inplace=inplace)
-        self.set_geometry(col, inplace=inplace)
+        if col in self.columns:
+            raise ValueError("Column named %s already exists" % col)
```

Comment:
Can you use either `.format()` syntax or f-string here?

Reply:
Of course